CRM/Core/Component.php

```diff
@@ -349,29 +349,39 @@ public static function searchAction(&$row, $id) {
   }
 
   /**
+   * Unused function.
+   *
    * @return array|null
+   *
+   * @deprecated
    */
-  public static function &contactSubTypes() {
+  public static function contactSubTypes() {
+    CRM_Core_Error::deprecatedWarning('unused');
```

Comment:
```
[bknix-min:~/bknix/build/universe/civicrm-core] grep -r '::contactSubTypes'
CRM/Core/Component.php:    $properties = self::contactSubTypes();
CRM/Custom/Page/Group.php:    $cSubTypes = CRM_Core_Component::contactSubTypes();
CRM/Custom/Form/Group.php:    $cSubTypes = CRM_Core_Component::contactSubTypes();
```

The latter two look live to me. I don't see any other callers in `universe`.

Both appear to be read-only consumers (ie :+1: for dropping `&`).

Reply:
I added that deprecation at the end & it looks like it was a bit aggressive - but since it came up I looked into the calls. Obviously the first is unused/deprecated. The second 2 are on browse & edit custom groups. However, stepping through the code I found the function only ever returned an empty array (which is logical as nothing sets those values) and obviously it works fine without that function returning anything - so I pushed up fixes to stop calling it